### PR TITLE
Create overlays for new kflux-ocp-p01 cluster

### PR DESCRIPTION
**Pull request by hugares**
This new cluster is an internal cluster similar to stone-prod-p02 but dedicated to the OCP (ART) team/product.

[KFLUXINFRA-921](https://issues.redhat.com//browse/KFLUXINFRA-921)

### CODE REVIEW

File: components/backup/production/kflux-ocp-p01/backup-s3-credentials-patch.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- op: replace
+  path: /spec/dataFrom/0/extract/key
+  value: integrations-output/terraform-resources/appsrep09ue1/stonesoup-infra-prod-backup/backup-kflux-ocp-p01
```
Comment (hugares) [inline components/backup/production/kflux-ocp-p01/backup-s3-credentials-patch.yaml:3]: @manish-jangra Did you create the s3 and kms in prod2 account?

File: components/backup/production/kflux-ocp-p01/dpa-kmskeyid-patch.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- op: replace
+  path: /spec/backupLocations/0/velero/config/kmsKeyId
+  value: 8361cc78-2507-4c79-a23b-346aa80bbcaf
```
Comment (hugares) [inline components/backup/production/kflux-ocp-p01/dpa-kmskeyid-patch.yaml:3]: To update with the kflux-ocp-p01 kms key once we have it

File: components/build-service/production/kflux-ocp-p01/pipelines-as-code-secret-path.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+- op: add
+  path: /spec/dataFrom/0/extract/key
+  value: production/build/kflux-ocp-p01/github-app
```
Comment (hugares) [inline components/build-service/production/kflux-ocp-p01/pipelines-as-code-secret-path.yaml:4]: This secret does not exist, we need to create the GH app, then secret

File: components/build-service/production/kflux-ocp-p01/webhook-config.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+    "https://github.com": "https://smee-smee.apps.stone-prd-host1.wdlc.p1.openshiftapps.com/redhathook13",
+    "https://gitlab.com": "https://smee-smee.apps.stone-prd-host1.wdlc.p1.openshiftapps.com/redhathook13"
```
Comment (hugares) [inline components/build-service/production/kflux-ocp-p01/webhook-config.json:3]: @gbenhaim This is a copy of p02, do we need to change something here?

File: components/integration/production/kflux-ocp-p01/pipelines-as-code-secret-path.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+- op: add
+  path: /spec/dataFrom/0/extract/key
+  value: production/pipeline-service/kflux-ocp-p01/github-app
```
Comment (hugares) [inline components/integration/production/kflux-ocp-p01/pipelines-as-code-secret-path.yaml:4]: This secret does not exist, we need to create the GH app, then secret

File: components/pipeline-service/production/kflux-ocp-p01/resources/update-tekton-config-pac.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+---
+- op: add
+  path: /spec/platforms/openshift/pipelinesAsCode/settings
+  value:
+    application-name: Konflux OCP
+    custom-console-name: Konflux OCP
```
Comment (hugares) [inline components/pipeline-service/production/kflux-ocp-p01/resources/update-tekton-config-pac.yaml:6]: @gbenhaim How should we name this one?

FYI:
p02 is "Konflux production internal"
rh01 is Red Hat Konflux"

File: components/backup/production/kflux-ocp-p01/backup-s3-credentials-patch.yaml (hunk 1 above)
Comment (manish-jangra) [reply components/backup/production/kflux-ocp-p01/backup-s3-credentials-patch.yaml:3]: not yet, I will do it today.

Comment (manish-jangra) [reply components/backup/production/kflux-ocp-p01/backup-s3-credentials-patch.yaml:3]: Merge Requested submitted.

**Review by manish-jangra — APPROVED** [review]
looks good to me.

We can wait for the GitLab MR to be merged.

File: components/build-service/production/kflux-ocp-p01/webhook-config.json (hunk 4 above)
Comment (gbenhaim) [reply components/build-service/production/kflux-ocp-p01/webhook-config.json:3]: yes we would need to use another channel for the webhooks.
The channel is created automatically by the client, so the only thing we need is to change the path in the url, for example `https://smee-smee.apps.stone-prd-host1.wdlc.p1.openshiftapps.com/**redhathook14**`

The same url should be included in the smee-client component.

File: components/pipeline-service/production/kflux-ocp-p01/resources/update-tekton-config-pac.yaml (hunk 6 above)
Comment (gbenhaim) [reply components/pipeline-service/production/kflux-ocp-p01/resources/update-tekton-config-pac.yaml:6]: I think `Konflux OCP` is fine. It's the name of the app that appears in the github status check.

**Review by manish-jangra — APPROVED** [review]
/approve
/lgtm

File: components/backup/production/kflux-ocp-p01/dpa-kmskeyid-patch.yaml (hunk 2 above)
Comment (manish-jangra) [reply components/backup/production/kflux-ocp-p01/dpa-kmskeyid-patch.yaml:3]: Here's the key - `5709bb39-3726-4a6c-aa1b-e5155cf981c1`

Comment (hugares) [reply components/backup/production/kflux-ocp-p01/dpa-kmskeyid-patch.yaml]: Done

File: components/build-service/production/kflux-ocp-p01/webhook-config.json (hunk 4 above)
Comment (hugares) [reply components/build-service/production/kflux-ocp-p01/webhook-config.json]: Done

**Pull request merged**